packages/framework/src/Model/Product/Recalculation/ProductRecalculationFacade.php

```diff
@@ -128,4 +150,27 @@ protected function groupProductIdsWithSameScopes(array $exportScopesIndexedByPro
 
         return $result;
     }
+
+    /**
+     * @param int[] $mainVariantsIds
+     * @return int[]
+     */
+    public function getIdsToRecalculate(array $mainVariantsIds): array
```

Comment:
if I dispatch `./bin/console shop:dis:rec 82 69` (both main variants), nothing is exported to elasticsearch. 

Also, this method seems really complex.

This method should return proper ids to recalculate - so it should return all variants, main variants and regular products?
if that's the case, I don't think why selecting by `id` or `main_variant_id` should not work 🤔  let's discuss it please